### ISSUE TITLE
Failing to train a model based on base-model and replace only tagger/parser.

### ISSUE THREAD

**Issue by aryehgigi** [explosion/spaCy#5160]
Hello spacy-ers,
I wanted to train a new model using a base model, while actully replacing both its tagger and parser, but keep the NER component. but with no luck,

## How to reproduce the behaviour
`python -m spacy train en models ./train.json ./dev.json -b en_core_web_sm -R -p 'tagger,parser'
`

but failed with (full traceback at the bottom):

> KeyError: "[E002] Can't find factory for ''tagger'. This usually happens when spaCy calls `nlp.create_pipe` with a component name that's not built in - for example, when constructing the pipeline from a model's meta.json. If you're using a custom component, you can write to `Language.factories[''tagger']` or remove it from the model meta and add it via `nlp.add_pipe` instead."

if I use only the -R flag without the -p, it would replace the original NER component which i wish to use, and if I use the -p without the -R I get the same exception and it says that it is adding and not replacing the tagger and parsers.

thanks
Aryeh

## Your Environment
* Operating System: Windows-10-10.0.17134-SP0
* Python Version Used: 3.7.6
* spaCy Version Used: 2.2.4
* Environment Information:

## Full traceback
Training pipeline: ["'tagger", "parser'"]
Starting with base model 'en_core_web_sm'
Adding component to base model ''tagger'
Traceback (most recent call last):
  File "c:\python37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\python37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "c:\python37\lib\site-packages\spacy\__main__.py", line 33, in <module>
    plac.call(commands[command], sys.argv[1:])
  File "c:\python37\lib\site-packages\plac_core.py", line 367, in call
    cmd, result = parser.consume(arglist)
  File "c:\python37\lib\site-packages\plac_core.py", line 232, in consume
    return cmd, self.func(*(args + varargs + extraopts), **kwargs)
  File "c:\python37\lib\site-packages\spacy\cli\train.py", line 204, in train
    nlp.add_pipe(nlp.create_pipe(pipe, config=pipe_cfg))
  File "c:\python37\lib\site-packages\spacy\language.py", line 300, in create_pipe
    raise KeyError(Errors.E002.format(name=name))
KeyError: "[E002] Can't find factory for ''tagger'. This usually happens when spaCy calls `nlp.create_pipe` with a component name that's not built in - for example, when constructing the pipeline from a model's meta.json. If you're using a custom component, you can write to `Language.factories[''tagger']` or remove it from the model meta and add it via `nlp.add_pipe` instead."

**Comment by adrianeboyd** [comment]
Hmm, I'm not able to replicate this. Can you try installing spacy in brand new virtual environment and running this command again?

Sometimes upgrading a python package doesn't clean up files from the previous version cleanly and it leads to weird errors.

**Comment by aryehgigi** [comment]
Thanks for the reply. I tried what you have suggested, and got the same result. But then I figured out the problem, my -p argument was wrong.
I wrote: 

> -p 'tagger,parser'

and should have written: 

> -p tagger,parser

this fixed it.

sorry for the confusion.
I think the confusion is because of the description in the training documentation on the -p option (https://spacy.io/api/cli#train): 

> Defaults to 'tagger,parser,ner'
so I thought the single quotes are needed 

thanks
